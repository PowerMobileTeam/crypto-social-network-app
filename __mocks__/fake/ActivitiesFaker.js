
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

