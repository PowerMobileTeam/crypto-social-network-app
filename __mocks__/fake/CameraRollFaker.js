
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

