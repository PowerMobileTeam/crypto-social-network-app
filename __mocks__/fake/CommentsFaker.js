
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

