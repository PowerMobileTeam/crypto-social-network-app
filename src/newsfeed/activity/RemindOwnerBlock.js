import React, {
  PureComponent
} from 'react';

import {
  Text,
  StyleSheet,
  Image,
  TouchableOpacity,
  View
} from 'react-native';

import { Icon } from 'react-native-elements';

import {
  NavigationActions
} from 'react-navigation';

import {
  MINDS_CDN_URI
} from '../../config/Config';

import {Â CommonStyle } from '../../styles/Common';

export default class RemindOwnerBlock extends PureComponent {


  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    // only active if NewsfeedList receive the navigation property
    if (this.props.navigation) {
      this.props.navigation.navigate('Channel', { guid:this.props.entity.ownerObj.guid, entity: this.props.entity.ownerObj });
    }
  }

  render() {
    const entity = this.props.entity.ownerObj;
    const avatarSrc = entity.getAvatarSource();
    return (
      <View style={styles.container}>
        <Icon color='rgb(70, 144, 214)' name='repeat' size={16} style={styles.icon}/>
        <TouchableOpacity onPress={this._navToChannel}>
          <Image source={avatarSrc} style={styles.avatar}/>
        </TouchableOpacity>
        <View style={styles.body}>
          <TouchableOpacity onPress={this._navToChannel} style={[CommonStyle.flexContainer, CommonStyle.columnAlignStart]}>
            <Text style={styles.username}>
              { entity.username }
            </Text>
            { this.props.entity.boosted &&
              <View style={styles.boostTagContainer}>
                <View style={styles.boostTagColumn}>
                  <Icon name="md-trending-up" type='ionicon' size={13} iconStyle={styles.boostTagIcon} />
                </View>
                <View style={styles.boostTagColumn}>
                  <Text style={styles.boostTagLabel}>BOOSTED</Text>
                </View>
              </View>
            }
          </TouchableOpacity>
        </View>
      </View>
    );
  }


}

const styles = StyleSheet.create({
  boostTagContainer: {
    flexDirection: 'row',
    flex:1,
    paddingRight: 32
  },
  boostTagColumn: {
    flexDirection: 'column',
    justifyContent: 'center',
  },
  boostTagIcon: {
    color: '#aaa',
  },
  boostTagLabel: {
    color: '#aaa',
    fontWeight: '800',
    marginLeft: 2,
    fontSize: 10,
  },
  container: {
    display: 'flex',
    flex: 1,
    paddingLeft: 8,
    paddingTop: 8,
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    marginLeft: 8,
    marginRight: 8,
  },
  avatar: {
    marginLeft: 8,
    height: 24,
    width: 24,
    borderRadius: 12,
    borderWidth: 0.5,
    borderColor: '#EEE',
  },
  body: {
    marginLeft: 8,
    flex:1
  },
  username: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#444',
    fontSize: 13,
  },
});
