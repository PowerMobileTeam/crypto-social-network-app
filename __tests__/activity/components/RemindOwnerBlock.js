import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import RemindOwnerBlock from '../../../src/newsfeed/activity/RemindOwnerBlock';

import formatDate from '../../../src/common/helpers/date';
import domain from '../../../src/common/helpers/domain';

describe('Remind owner component', () => {

  let user, comments, entity, screen;
  beforeEach(() => {
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <RemindOwnerBlock entity={activityResponse.activities[0]}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have Touchableopacity', async () => {
    screen.update();
    expect(screen.find(TouchableOpacity)).toHaveLength(2);
  });

  
  it('should _navToChannel on press ', () => {
    let activityResponse = activitiesServiceFaker().load(1);

    const navigation = { 
      navigate: jest.fn() 
    };
    let entity = activityResponse.activities[0];
    screen = shallow(
      <RemindOwnerBlock entity={entity} navigation={navigation} rightToolbar={null}/>
    );
    screen.update()
    let render = screen.dive();
    let touchables = render.find('TouchableOpacity');
    touchables.at(0).props().onPress();
    jest.runAllTimers();

    expect(navigation.navigate).toHaveBeenCalledWith('Channel', {'entity': entity.ownerObj, 'guid': entity.ownerObj.guid});


    expect(screen.find(TouchableOpacity)).toHaveLength(2);

  });

});
