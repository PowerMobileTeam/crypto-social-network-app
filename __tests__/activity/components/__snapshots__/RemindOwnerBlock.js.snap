// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remind owner component renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
      "paddingLeft": 8,
      "paddingTop": 8,
    }
  }
>
  <Icon
    color="rgb(70, 144, 214)"
    name="repeat"
    raised={false}
    reverse={false}
    reverseColor="white"
    size={16}
    style={
      Object {
        "marginLeft": 8,
        "marginRight": 8,
      }
    }
    underlayColor="white"
  />
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Image
      source={
        Object {
          "source": "http://thisisaurl",
        }
      }
      style={
        Object {
          "borderColor": "#EEE",
          "borderRadius": 12,
          "borderWidth": 0.5,
          "height": 24,
          "marginLeft": 8,
          "width": 24,
        }
      }
    />
  </TouchableOpacity>
  <Component
    style={
      Object {
        "flex": 1,
        "marginLeft": 8,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "alignItems": "flex-start",
            "flexDirection": "column",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#444",
            "fontFamily": "Roboto",
            "fontSize": 13,
            "fontWeight": "bold",
          }
        }
      />
    </TouchableOpacity>
  </Component>
</Component>
`;
